src/ops/PromoteOps.ts

```diff
@@ -0,0 +1,1217 @@
+ * @param keysToRemove keys to remove from the objects before comparing them
+ */
+function removeKeysAndCompare(
+  masterFilePath: string,
+  counterpartPath: string,
+  keysToRemove: Array<string>
+): boolean {
+  const data = fs.readFileSync(masterFilePath, 'utf8');
+  const obj = removeKeys(JSON.parse(data), keysToRemove);
+  const dataCopy = fs.readFileSync(counterpartPath, 'utf8');
+  const objCopy = removeKeys(JSON.parse(dataCopy), keysToRemove);
+  return JSON.stringify(objCopy) === JSON.stringify(obj);
+}
+
+/**
+ * removes specific keys and values from json objects
+ * @param obj the object that we want to remove the values from
+ * @param keysToRemove the keys to remove from that object
+ */
+function removeKeys(obj, keysToRemove) {
+  return Object.fromEntries(
+    Object.entries(obj)
+      .filter(([key]) => !keysToRemove.includes(key)) // Exclude specified keys
+      .map(([key, value]) =>
+        value && typeof value === 'object'
+          ? [key, removeKeys(value, keysToRemove)]
+          : [key, value]
+      )
+  );
+}
+
+/**
+ * removes nulls from json objects and then compares the two objects for changes
+ * @param masterFilePath path to the master config object
+ * @param counterpartPath path to the export config object
+ */
+function removeNullsAndCompare(
+  masterFilePath: string,
+  counterpartPath: string
+): boolean {
+  const data = fs.readFileSync(masterFilePath, 'utf8');
+  const object = removeNulls(JSON.parse(data));
+  const dataCopy = fs.readFileSync(counterpartPath, 'utf8');
+  const objectCopy = removeNulls(JSON.parse(dataCopy));
+  return JSON.stringify(object) === JSON.stringify(objectCopy);
+}
+
+/**
+ * removes nulls from the config objects
+ * @param obj object to remove the null values from
+ */
+function removeNulls(obj) {
+  return Object.fromEntries(
+    Object.entries(obj)
+      .filter(([, value]) => value !== null)
+      .map(([key, value]) =>
+        value && typeof value === 'object'
+          ? [key, removeNulls(value)]
+          : [key, value]
+      )
+  );
+}
+
+/**
+ * When these objects change certain parts are not as important as others when they are changed
+ * so if that part is determined to have been the part that changed and other parts did not change
+ * the object will not be flagged as changed.
+ * @param path sub-path to the config object
+ * @param dir path to the master directory
+ * @param counterpartPath path to the export directory
+ */
+function checkChange(path: string, dir: string, counterpartPath: string) {
+  const type = getTypeFromPath(path);
+  const importFilePath = dir + '/' + path;
+  switch (type) {
+    case 'policy': {
+      const keysToRemove = [
+        'createdBy',
+        'creationDate',
+        'lastModifiedDate',
+        'lastModifiedBy',
+      ];
+      if (removeKeysAndCompare(importFilePath, counterpartPath, keysToRemove)) {
+        return;
+      }
+      break;
+    }
+    case 'resourcetype': {
+      const keysToRemove = [
```

Comment:
Do we just remove the metadata instead of individual keys?

Reply:
We could make a method that does something similar to that but these export with the metadata even when the flag is there to remove the metadata because I think they are part of the object in a different way

Reply:
Yes, when the exports are done, some objects do have the metadata as part of them. The saveToFile and saveJsonToFile functions that save the export are what eventually remove it (if it exists) or add them (if it doesn't exist) to the export when it is written to a file. You can simply do `delete obj.meta` to delete the metadata if it exists. However, I realized these keys might be attributes of the object instead and not part of the meta object, in which case we probably do need to have it remove the individual keys if we want to ignore those.